adapters/sn2core/sn2core.go

```diff
@@ -157,15 +157,32 @@ func AdaptTransaction(transaction *starknet.Transaction) (core.Transaction, erro
 
 func AdaptDeclareTransaction(t *starknet.Transaction) *core.DeclareTransaction {
 	return &core.DeclareTransaction{
-		TransactionHash:      t.Hash,
-		SenderAddress:        t.SenderAddress,
-		MaxFee:               t.MaxFee,
-		TransactionSignature: *t.Signature,
-		Nonce:                t.Nonce,
-		Version:              (*core.TransactionVersion)(t.Version),
-		ClassHash:            t.ClassHash,
-		CompiledClassHash:    t.CompiledClassHash,
+		TransactionHash:       t.Hash,
+		SenderAddress:         t.SenderAddress,
+		MaxFee:                t.MaxFee,
+		TransactionSignature:  *t.Signature,
+		Nonce:                 t.Nonce,
+		Version:               (*core.TransactionVersion)(t.Version),
+		ClassHash:             t.ClassHash,
+		CompiledClassHash:     t.CompiledClassHash,
+		ResourceBounds:        adaptResourceBounds(t.ResourceBounds),
+		Tip:                   safeFeltToUint64(t.Tip),
+		PaymasterData:         t.PaymasterData,
+		AccountDeploymentData: t.AccountDeploymentData,
+		NonceDAMode:           core.DataAvailabilityMode(t.NonceDAMode),
+		FeeDAMode:             core.DataAvailabilityMode(t.FeeDAMode),
+	}
+}
+
+func adaptResourceBounds(rb map[starknet.Resource]starknet.ResourceBounds) map[core.Resource]core.ResourceBounds {
+	coreBounds := make(map[core.Resource]core.ResourceBounds, len(rb))
+	for resource, bounds := range rb {
+		coreBounds[core.Resource(resource)] = core.ResourceBounds{
+			MaxAmount:       bounds.MaxAmount.Uint64(),
```

Comment:
is it safe to assume `bounds.MaxAmount` fits in a uint64?

Reply:
It's what the spec says https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-8.md

Reply:
We will have to do db migrations if we wait. Might be better to delay the release by 1-2 days and just implement them the first time imho.

Reply:
> We will have to do db migrations if we wait
Not really, felts are stored as big-endian byte arrays. We can keep using the underlying data in the database, it is no big deal.